client/boot/common.js

```diff
@@ -137,14 +137,13 @@ const unsavedFormsMiddleware = () => {
 export const locales = ( currentUser, reduxStore ) => {
 	debug( 'Executing Calypso locales.' );
 
-	if ( window.i18nLocaleStrings ) {
-		const i18nLocaleStringsObject = JSON.parse( window.i18nLocaleStrings );
-		reduxStore.dispatch( setLocaleRawData( i18nLocaleStringsObject ) );
+	if ( window.i18nLocaleObject ) {
+		reduxStore.dispatch( setLocaleRawData( window.i18nLocaleObject ) );
 	}
 
 	// Use current user's locale if it was not bootstrapped (non-ssr pages)
 	if (
-		! window.i18nLocaleStrings &&
+		! window.i18nLocaleObject &&
```

Comment:
I'd consider `delete window.i18nLocaleObject` after we use the object to initialize the `i18n` library. We can free the memory allocated by the object instead of holding it indefinitely.

With an object, it probably doesn't matter that much (the object reference is passed to the `i18n` library and the memory is still used), but with the `i18nLocaleStrings` string, we were leaking 700kB of memory allocated for the serialized string.